PokeOneWeb.DataSync/ReadModelUpdate/Impl/ReadModelUpdateService.cs

```diff
@@ -100,7 +104,127 @@ public void UpdateReadModel(SpreadsheetImportReport importReport)
             _regionRepository.Update(_regionMapper.MapFromDatabase(importReport));
             _reporter.StopReadModelUpdate("regions");
 
+            _reporter.StartReadModelUpdate("generate-json-files");
+            GenerateJsonFiles(importReport);
+            _reporter.StopReadModelUpdate("generate-json-files");
+
             _reporter.StopReadModelUpdate();
         }
+
+        private void GenerateJsonFiles(SpreadsheetImportReport importReport)
+        {
+            CreateDirectories();
+
+            var serializeOptions = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                WriteIndented = true
+            };
+
+            //
+            // Entities
+            // 
+            Console.WriteLine("generating json files for entities");
+            ICollection<EntityTypeReadModel> entities = _entityTypeMapper.MapFromDatabase(importReport).Keys;
+            File.WriteAllText("resources/entities.json", JsonSerializer.Serialize(entities, serializeOptions));
+
+            foreach (var entity in entities)
+            {
+                File.WriteAllText("resources/entities/" + entity.ResourceName + ".json", JsonSerializer.Serialize(entity, serializeOptions));
+            }
+
+            //
+            // itemstats
+            // 
+            Console.WriteLine("generating json files for itemstats");
+            ICollection<ItemStatBoostPokemonReadModel> itemStats = _itemStatBoostPokemonMapper.MapFromDatabase(importReport).Keys;
+            File.WriteAllText("resources/itemstats.json", JsonSerializer.Serialize(itemStats, serializeOptions));
+
+            //
+            // learnable moves
+            //
+            Console.WriteLine("generating json files for learnable-moves");
+            ICollection<SimpleLearnableMoveReadModel> learnableMoves = _simpleLearnableMoveMapper.MapFromDatabase(importReport).Keys;
+            File.WriteAllText("resources/learnable-moves.json", JsonSerializer.Serialize(learnableMoves, serializeOptions));
+
+            foreach (var learnableMove in learnableMoves)
+            {
+                File.WriteAllText("resources/learnable-moves/" + learnableMove.MoveResourceName + ".json", JsonSerializer.Serialize(learnableMove, serializeOptions));
```

Comment:
Yes this is the best solution I think. One file per move. For AND queries, the files for all moves that are queried are loaded and compared to find Pokémon that can learn all these moves.